worker/raft/queue/blockingqueue_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package queue
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+type BlockingOpQueueSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&BlockingOpQueueSuite{})
+
+func (s *BlockingOpQueueSuite) TestEnqueue(c *gc.C) {
+	queue := NewBlockingOpQueue()
+
+	go func() {
+		for op := range queue.Queue() {
+			c.Assert(op.Command, gc.DeepEquals, []byte("abc"))
```

Comment:
I don't think `gc.C` is Goroutine-safe...

Reply:
`-race` didn't care, but I'll change it.